src/TwigComponent/src/Command/ComponentDebugCommand.php

```diff
@@ -0,0 +1,265 @@
+            )
+        ;
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $io = new SymfonyStyle($input, $output);
+        $name = $input->getArgument('name');
+        $componentsDir = $input->getOption('dir');
+
+        if (null !== $name) {
+            try {
+                $metadata = $this->componentFactory->metadataFor($name);
+            } catch (\Exception $e) {
+                $io->error($e->getMessage());
+
+                return Command::FAILURE;
+            }
+
+            $class = $metadata->get('class');
+            $type = null;
+            $allProperties = [];
+
+            if ($class) {
+                $type = 'AsTwigComponent';
+
+                if ($metadata->get('live')) {
+                    $type = 'AsLiveComponent';
+                }
+
+                $reflectionClass = new \ReflectionClass($class);
+                $properties = $reflectionClass->getProperties();
+                $allLiveProperties = [];
+
+                foreach ($properties as $property) {
+                    if ($property->isPublic()) {
+                        $visibility = $property->getType()?->getName();
+                        $propertyName = $property->getName();
+                        $value = $property->getDefaultValue();
+                        $propertyAttributes = $property->getAttributes(LiveProp::class);
+
+                        $propertyDisplay = $visibility.' $'.$propertyName.(null !== $value ? ' = '.$value : '');
+
+                        if (\count($propertyAttributes) > 0) {
+                            $allLiveProperties[] = $propertyDisplay;
+                        } else {
+                            $allProperties[] = $propertyDisplay;
+                        }
+                    }
+                }
+
+                $methods = $reflectionClass->getMethods();
+                $allEvents = [];
+                $allActions = [];
+
+                foreach ($methods as $method) {
+                    if ('mount' === $method->getName()) {
+                        $allEvents[] = 'Mount';
+                    }
+
+                    foreach ($method->getAttributes() as $attribute) {
+                        if (PreMount::class === $attribute->getName()) {
+                            $allEvents[] = 'PreMount';
+                            break;
+                        }
+
+                        if (PostMount::class === $attribute->getName()) {
+                            $allEvents[] = 'PostMount';
+                            break;
+                        }
+
+                        if (LiveAction::class === $attribute->getName()) {
+                            $allActions[] = $method->getName();
+                            break;
+                        }
+                    }
+                }
+            } else {
+                $source = $this->twigEnvironment->load($metadata->getTemplate())->getSourceContext();
```

Comment:
Can we extract this block to a `private getPropertiesForAnonymousComponent()` and the above block a `private `getPropertiesForClassComponent()`? Just for readability.